### PR TITLE
security: add commit hash to docker file

### PR DESCRIPTION
**Pull request by andife**
### Description
best practice (security) to add commit hash

### CODE REVIEW

**Comment by justinchuby** [comment]
Will this be auto updated by https://github.com/onnx/onnx/blob/main/.github/dependabot.yml ?

**Comment by andife** [comment]
> Will this be auto updated by https://github.com/onnx/onnx/blob/main/.github/dependabot.yml ?

not yet. I have to add an entry for docker there. Will update that.

**Comment by andife** [comment]
> > Will this be auto updated by https://github.com/onnx/onnx/blob/main/.github/dependabot.yml ?
> 
> not yet. I have to add an entry for docker there. Will update that.

added/updated the dependabot rules for docker

File: .github/workflows/release_linux_x86_64.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
         python -m pip install -q -r requirements-release.txt
 
     - name: Build manylinux2014_x86_64
-      uses: docker://quay.io/pypa/manylinux2014_x86_64
+      uses: docker://quay.io/pypa/manylinux2014_x86_64:latest@sha256:085b7c2c1bcb45b936e5478fe24a5142b0519e242a0104142fd50d569d6cb922
```
Comment (jcwchen) [inline .github/workflows/release_linux_x86_64.yml:46]: I am inclined to always use the latest docker image. Will [dependabot](https://github.com/apps/dependabot) also help us to bump the version if there is an update?

**Review by jcwchen — COMMENTED** [review]
https://github.com/onnx/onnx/blob/89a4eb071185244882e1bbd167ec384e9b8a07a4/.github/workflows/release_linux_aarch64.yml#L27 This line will need to be updated also?

File: .github/workflows/release_linux_x86_64.yml (hunk 1 above)
Comment (andife) [reply .github/workflows/release_linux_x86_64.yml:46]: I added a rule for docker at .github/dependabot.yml ... so it should work.

**Comment by jcwchen** [comment]
Shall we also update:
[[onnx/.github/workflows/release_linux_aarch64.yml](https://github.com/onnx/onnx/blob/89a4eb071185244882e1bbd167ec384e9b8a07a4/.github/workflows/release_linux_aarch64.yml#L27)](https://github.com/onnx/onnx/blob/89a4eb071185244882e1bbd167ec384e9b8a07a4/.github/workflows/release_linux_aarch64.yml#L27)
with specified commit hash?

File: .github/workflows/release_linux_aarch64.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
     env:
       # setting up python and docker image
       py: /opt/python/${{ matrix.python-version }}/bin/python
-      img: quay.io/pypa/manylinux2014_aarch64
+      img: docker://quay.io/pypa/manylinux2014_aarch64@a2d8ba969c23fb91e383b99f62324769944fe2518404b56b186037bab7ad1117
```
Comment (jcwchen) [inline .github/workflows/release_linux_aarch64.yml:27]:
```suggestion
      img: docker://quay.io/pypa/manylinux2014_aarch64:latest@sha256:a2d8ba969c23fb91e383b99f62324769944fe2518404b56b186037bab7ad1117
```

Comment (andife) [reply .github/workflows/release_linux_aarch64.yml]: maybe we ignore these Docker files in environments for now. Somehow the syntax doesn't seem to fit quite right. 
For the score at scorecard they don't seem to be detected, and then dependabot should also find them as stale. If one has the correct syntax, I would perhaps test this with the second last hash if we then get an automatic update suggestion.

**Review by jcwchen — APPROVED** [review]

**Pull request merged**